### PR TITLE
Feature/cleanup

### PR DESCRIPTION
**Pull request by YUzhva**
[Link to JIRA ticket](https://ssu-jira.softserveinc.com/browse/LR-)

### Author.

- [ ] I have added link to this PR to the JIRA
- [ ] I have logged time in JIRA.

### Reviewers.

- [ ] @reviewer1
- [ ] @reviewer2
- [ ] @vovka

### Issue summary.

*Short description of you feature or bug.*

### Fix summary. 

_Short description of how you implemented a feature or fixed a bug. And why._

### Testing approach.

_Please describe your testing approach here._
* _What cases do your tests cover?_
* _Did you cover all of the edge cases?_
* _What kinds of tests did you write?_
  * _unit tests?_
  * _controller tests?_
  * _integration tests?_
* _What was your rationale for choosing one kind of test over another for testing each case?_

### CODE REVIEW

File: app/controllers/interviews_controller.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  end
+
+  def show
+  end
+
+  private
+
+  def interview_params
+    @first_question = @quiz.questions.first
+    params.permit.merge(quiz_id: @quiz.id, user_id: current_user.id, current_question_id: @first_question.id)
```
Comment (vovka) [inline app/controllers/interviews_controller.rb]: Codacy Issue found: [Line is too long. [109/80]](https://www.codacy.com/app/vovka/video_quiz/file/2613079628/issues/source?bid=3205144&fileBranchId=3311714#l26)

File: app/controllers/interviews_controller.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+class InterviewsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_quiz
+  
+  def new
+  end
+
+  def create
```
Comment (vovka) [inline app/controllers/interviews_controller.rb]: Codacy Issue found: [Method has too many lines. [7/5]](https://www.codacy.com/app/vovka/video_quiz/file/2613079628/issues/source?bid=3205144&fileBranchId=3311714#l8)

File: app/controllers/interviews_controller.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    @interview = @quiz.check_interview(current_user, interview_params)
+
+    respond_to do |format|
+      if @interview.save
+        format.html { redirect_to new_quiz_answer_path(@quiz), notice: 'Interview was successfully created.' }
```
Comment (vovka) [inline app/controllers/interviews_controller.rb]: Codacy Issue found: [Line is too long. [110/80]](https://www.codacy.com/app/vovka/video_quiz/file/2613079628/issues/source?bid=3205144&fileBranchId=3311714#l13)

File: app/controllers/interviews_controller.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  private
+
+  def interview_params
+    @first_question = @quiz.questions.first
+    params.permit.merge(quiz_id: @quiz.id, user_id: current_user.id, current_question_id: @first_question.id)
+  end
+
+  def set_quiz
+    @quiz = Quiz.find(params[:quiz_id])
```
Comment (vovka) [inline app/controllers/interviews_controller.rb]: Codacy Issue found: [Unscoped call to Quiz#find](https://www.codacy.com/app/vovka/video_quiz/file/2613079628/issues/source?bid=3205144&fileBranchId=3311714#l30)

File: app/assets/javascripts/angularjs_app/components/home/homeControllers.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+var homeControllers = angular.module('homeControllers', []);
```
Comment (vovka) [inline app/assets/javascripts/angularjs_app/components/home/homeControllers.js]: Codacy Issue found: ["angular" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2613079685/issues/source?bid=3205144&fileBranchId=3311714#l1)

File: spec/requests/answers_spec.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe "Answers", type: :request do
+  before(:each) do
+    @user = User.create!(email: "user@example.org", name: "Name")
+    sign_in @user
+    @quiz = Quiz.create!(name: 'This is quiz #1')
+    @interview = Interview.create!(quiz_id: 1, user_id: 1, current_question_id: 1)
+    @quiz.questions.create!(content: 'This is question #1', record_time_limit: 10)
```
Comment (vovka) [inline spec/requests/answers_spec.rb]: Codacy Issue found: [Line is too long. [82/80]](https://www.codacy.com/app/vovka/video_quiz/file/2613079672/issues/source?bid=3205144&fileBranchId=3311714#l9)

File: spec/requests/answers_spec.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe "Answers", type: :request do
+  before(:each) do
+    @user = User.create!(email: "user@example.org", name: "Name")
+    sign_in @user
+    @quiz = Quiz.create!(name: 'This is quiz #1')
+    @interview = Interview.create!(quiz_id: 1, user_id: 1, current_question_id: 1)
```
Comment (vovka) [inline spec/requests/answers_spec.rb]: Codacy Issue found: [Line is too long. [82/80]](https://www.codacy.com/app/vovka/video_quiz/file/2613079672/issues/source?bid=3205144&fileBranchId=3311714#l8)

File: db/schema.rb [hunk 8]
```diff
@@ -11,7 +11,31 @@
...
+    t.integer  "quiz_id"
+    t.integer  "user_id"
+    t.integer  "current_question_id"
+    t.string   "status"
+    t.datetime "created_at",          null: false
+    t.datetime "updated_at",          null: false
+  end
+
+  add_index "interviews", ["current_question_id"], name: "index_interviews_on_current_question_id"
+  add_index "interviews", ["quiz_id", "user_id"], name: "index_interviews_on_quiz_id_and_user_id", unique: true
```
Comment (vovka) [inline db/schema.rb]: Codacy Issue found: [Line is too long. [111/80]](https://www.codacy.com/app/vovka/video_quiz/file/2613079659/issues/source?bid=3205144&fileBranchId=3311714#l36)

File: db/schema.rb [hunk 9]
```diff
@@ -11,7 +11,31 @@
...
+  create_table "interviews", force: :cascade do |t|
+    t.integer  "quiz_id"
+    t.integer  "user_id"
+    t.integer  "current_question_id"
+    t.string   "status"
+    t.datetime "created_at",          null: false
+    t.datetime "updated_at",          null: false
+  end
+
+  add_index "interviews", ["current_question_id"], name: "index_interviews_on_current_question_id"
```
Comment (vovka) [inline db/schema.rb]: Codacy Issue found: [Line is too long. [98/80]](https://www.codacy.com/app/vovka/video_quiz/file/2613079659/issues/source?bid=3205144&fileBranchId=3311714#l35)

File: spec/models/quiz_spec.rb [hunk 10]
```diff
@@ -32,4 +35,51 @@
...
+    @quiz.questions.create!(content: 'This is question #2', record_time_limit: 20)
+    @interview = Interview.create(quiz_id: 1, user_id: 1, current_question_id: 1, status: nil)
+
+    expect(@quiz.next_quiz_question(@interview.current_question_id).id).to eq(2)
+  end
+
+  describe 'interviews' do
+    before {
+      @quiz.save
+      @user.save
```
Comment (vovka) [inline spec/models/quiz_spec.rb]: Codacy Issue found: [Line is too long. [101/80]](https://www.codacy.com/app/vovka/video_quiz/file/2613079670/issues/source?bid=3205144&fileBranchId=3311714#l52)
